LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/JoinRequestMessageHandler.cs

```diff
@@ -154,7 +154,10 @@ async Task ProcessJoinRequestAsync(LoRaRequest request)
                 if (windowToUse == Constants.RECEIVE_WINDOW_1)
                 {
                     datr = loraRegion.GetDownstreamDR(request.Rxpk);
-                    freq = loraRegion.GetDownstreamChannelFrequency(request.Rxpk);
+                    if (!loraRegion.TryGetDownstreamChannelFrequency(request.Rxpk, out freq) || datr == null)
```

Comment:
that's a bit weird. If you do not get a valid frequency, can you continue?

Reply:
Thank you for the comment, this is a bug. The same occur inDefaultDataRequestHandler, I tried to correct it